src/ffi/dial_ffi.rs

```diff
@@ -137,6 +141,21 @@ pub unsafe extern "C" fn dial(
         ur
     };
     let allow_insec = c_allow_insec;
+    if!c_payload.is_null() && c_type.is_null(){
+        println!("Error missing credential: secret type");
+        return ptr::null_mut();
+
+    }
+    if c_payload.is_null() && !c_type.is_null(){
```

Comment:
These checks feel a bit redundant given what's happening below at line 202. And, though we define `r#type` and `payload` here, they don't actually get used until line 202. Can we just combine all this logic at a single point so we're not doing this "is it null?" check and also a "is it None?" check?